### DISCUSSION TITLE
Assigning ID value to new sample providers in hospitals.csv

### DISCUSSION THREAD

**Discussion by npagare** [synthetichealth/synthea#1504] [General]
Hello,

Referring to the wiki - https://github.com/synthetichealth/synthea/wiki/Provider-Data, the hospitals.csv doesn't have - 
> - first column with no header | (I see that as a provider_num)
> - second column is no id | (I see that as npi)

So, which of the provider_num and/or npi columns have values assigned by CMS?

Is there any guidance on how to create and assign a value to the provider_num and npi fields, if I were to create a new sample provider entry in the hospitals.csv? Is there other file I should update for such new provider entry?

What do I need to add/change to associate this provider with a new Organization?

Thank you

**Comment by jawalonoski** [comment]
It looks like that wiki page needs an update.

Various files in the `src/main/resources/providers` folder have slightly different formats. The files are read by _column name_, not column order. You can see the logic starting here: https://github.com/synthetichealth/synthea/blob/995cf2fd33e67918d4e33110d9f68ad248002221/src/main/java/org/mitre/synthea/world/agents/Provider.java#L670

The `"provider num"` column is from CMS, and in some non-Synthea systems or files it appears as `PRVDR_NUM`.

The `"npi"` column is only for facility or organizational NPIs, and it comes from CMS when it exists.

If you don't intend to interface with CMS data or systems, you don't have to maintain those values, and you can do whatever you want with them.

The only internal restriction is that the combination of `npi + name` is used to construct a `UUID` for each provider, and that UUID must be unique.

**Comment by npagare** [comment]
@jawalonoski, thank you for the guidance.
My apologies for the delayed response.

Any thoughts on how to make custom npi unique enough to not to collide with CMS assigned number?
Will alpha-numeric value be, ok?

Thank you

**Reply by jawalonoski** [reply]
The NPI format is documented https://www.cms.gov/Regulations-and-Guidance/Administrative-Simplification/NationalProvIdentStand/Downloads/NPIcheckdigit.pdf

I can provide no other guidance or suggestions.

**Reply by hadleynet** [reply]
This code might be of use: https://github.com/synthetichealth/synthea/blob/76f2925bbcb098c34413b4c1e245d8664d7251d7/src/main/java/org/mitre/synthea/world/agents/Provider.java#L636

**Reply by npagare** [reply]
@jawalonoski  and @hadleynet - thank you both for your help and guidance. Much appreciated!
